xapian-applications/omega/templates/query

```diff
@@ -108,9 +108,9 @@ $if{$field{size},<span title="$html{$field{size}} bytes">Size: <b>$html{$filesiz
 </div>
 </div>
 </td>
-<td><b><a href="$if{$opt{clicklog},$html{$or{$env{SCRIPT_NAME},omega.cgi}?FMT=clicklog&DB=$url{$dbname}&QUERYID=$opt{queryid}&DOCID=$url{$id}&URL=$url{$opt{docurl}}},$html{$opt{docurl}}}">$html{$or{$field{caption},$field{title},$prettyurl{$opt{docurl}},Untitled}}</a></b><br>
+<td><b><a href="$if{$opt{clicklog},$opt{clicklogurl},$html{$field{url}}}">$html{$or{$field{caption},$field{title},$prettyurl{$field{url}},Untitled}}</a></b><br>
 <small>$snippet{$field{sample}}</small><br>
-<a href="$if{$opt{clicklog},$html{$or{$env{SCRIPT_NAME},omega.cgi}?FMT=clicklog&DB=$url{$dbname}&QUERYID=$opt{queryid}&DOCID=$url{$id}&URL=$url{$opt{docurl}}},$html{$opt{docurl}}}">$html{$opt{docurl}}</a><br>
+<a href="$if{$opt{clicklog},$opt{clicklogurl},$html{$field{url}}}">$html{$field{url}}</a><br>
```

Comment:
ARGH!  This is still unhelpfully duplicating code - we still have two copies of this code to decide which URL to insert:

```
$if{$opt{clicklog},$opt{clicklogurl},$html{$field{url}}}
```

What I'm suggesting should go in the option is the URL which we want in both these links.  So the second use should look like:

```
<a href="$html{$opt{clickurl}}">$html{$field{url}}</a>
```

Here `$opt{clickurl}` is the URL the user gets to click on and `$field{url}` which is the URL of the resource itself.  In the default case the two are the same, but in the "clicklog" case the "clickurl" takes us via the redirect.

I'd suggest keeping the `$html{...}` escaping step at the point of use, as that gives a cleaner separation between unescaped and escaped data.  If some data is already escaped and some isn't, it's harder to follow where escaping is needed, which means more of a breeding ground for bugs.